Update README.MD to explain fragment locations (#6)
Ibotta/retrofit-graphql: README.md

### The Basics

> Firstly you'll need somewhere to save your GraphQL queries, fragments, and mutations into .graphql files. You can use this tool to generate your Insomnia workspaces into directories and files [insomnia-graphql-generator](https://github.com/AniTrend/insomnia-graphql-generator) and place these files into your assets folder as shown below:

Reviewer: ```suggestion
First, you'll need .graphql files for saving your GraphQL queries, fragments, and mutations into. You can use this tool to generate your Insomnia workspaces into directories and files [insomnia-graphql-generator](https://github.com/AniTrend/insomnia-graphql-generator) and place them into your assets folder as shown below:
```

<img src="./images/screenshots/assets_files.png" width=250 />

> With respect to fragments, you have two options. Either place them inside your query file after the query definition or put them in their own file under the `assets/graphql/Example/Fragment/` folder. You may also use a mixture of the two if you wish. Note, only define one fragment per file and make sure the name of the fragment matches the filename. For example:

Reviewer: ```suggestion
With respect to fragments, you have two options. Either place them inside your query file after the query definition, or put them in their own file under the `assets/graphql/Example/Fragment/` folder. You may also use a mixture of the two if you wish. Note, only define one fragment per file and make sure the name of the fragment matches the filename. Here's an example:
```

> With respect to fragments, you have two options as to where you can place them. You can place them inside your query file after the query definition. Or, you can put your fragments in their own file under the `assets/graphql/Example/Fragment/` folder. You can also use a mix of the two if you wish. Only define one fragment per file, and make sure the name of the fragment matches the filename. For example:

Reviewer: ```suggestion
With respect to fragments, you have two options. You can place them inside your query file after the query definition. Or, you can put your fragments in their own file under the `assets/graphql/Example/Fragment/` folder. You may also use a mix of the two if you wish. Note, only define one fragment per file and make sure the name of the fragment matches the filename. For example:
```

> With respect to fragments, you have two options as to where you can place them. You can place them inside your query file, after the query definition. Or, you can put your fragments in their own file under the `assets/graphql/Example/Fragment/` folder. You can also use a mix of the two if you wish. Only define one fragment per file, and make sure the name of the fragment matches the filename. For example:

Reviewer: ```suggestion
With respect to fragments, you have two options as to where you can place them. You can place them inside your query file after the query definition. Or, you can put your fragments in their own file under the `assets/graphql/Example/Fragment/` folder. You can also use a mix of the two if you wish. Only define one fragment per file, and make sure the name of the fragment matches the filename. For example:
```

> In the file `assets/graphql/Example/Query/Trending.graphql` two fragments (`RepositoryFragment` and `UserFragment`) are referenced but not defined in the same file.

Reviewer: ```suggestion
In the file `assets/graphql/Example/Query/Trending.graphql`, two fragments (`RepositoryFragment` and `UserFragment`) are referenced but not defined in the same file.
```

> In the example file `assets/graphql/Example/Query/Trending.graphql` two fragments (`RepositoryFragment` and `UserFragment`) are referenced but not defined in the same file.

Reviewer: ```suggestion
In the file `assets/graphql/Example/Query/Trending.graphql` two fragments (`RepositoryFragment` and `UserFragment`) are referenced but not defined in the same file.
```

> A query that lives in: `assets/graphql/Example/Query/Trending.graphql`. It references two fragments (`RepositoryFragment` and `UserFragment`) that were not defined with the query in the same file. They fragment definitions live elsewhere.

Reviewer 2: > _The_ fragment definitions live elsewhere

> A query that lives in: `assets/graphql/Example/Query/Trending.graphql`. It references two fragments (`RepositoryFragment` and `UserFragment`) that were not defined with the query in the same file. They fragment definitions live elsewhere.

Reviewer: ```suggestion
In the example file `assets/graphql/Example/Query/Trending.graphql` two fragments (`RepositoryFragment` and `UserFragment`) are referenced but not defined in the same file.
```

> A query that lives in: `assets/graphql/Example/Query/Trending.graphql`. It references two fragments (`RepositoryFragment` and `UserFragment`) that were not defined with the query in the same file. They fragment definitions live elsewhere.

Reviewer 2: ```suggestion
A query that lives in: `assets/graphql/Example/Query/Trending.graphql`. It references two fragments (`RepositoryFragment` and `UserFragment`) that were not defined with the query in the same file. The fragment definitions live elsewhere.
```

```
query Trending($type: FeedType!, $offset: Int, $limit: Int) {
  feed(type: $type, offset: $offset, limit: $limit) {
    id
    hotScore
    repository {
      ...RepositoryFragment
    }
    postedBy {
      ...UserFragment
    }
  }
}
```

> The `RepositoryFragment` lives in `assets/graphql/Example/Fragment/RepositoryFragment.graphql`. It happens to reference a `UserFragment` as well, which is defined in its own file. It is fine for fragments to reference other fragments:

Reviewer: ```suggestion
The `RepositoryFragment` lives in `assets/graphql/Example/Fragment/RepositoryFragment.graphql`. It happens to reference `UserFragment` which is also defined in its own file.
```

Author: This part is important ` It is fine for fragments to reference other fragments`. It calls out that you can recursively refer to fragments

Reviewer: Oh word. Fixing.
